runtime/protocol/http-client-engines/http-client-engine-okhttp/jvm/src/aws/smithy/kotlin/runtime/http/engine/okhttp/OkHttpEngine.kt

```diff
@@ -70,20 +70,29 @@ private fun OkHttpEngineConfig.buildClient(): OkHttpClient {
         followSslRedirects(false)
 
         // see: https://github.com/ktorio/ktor/issues/1708#issuecomment-609988128
+        // TODO disable this once better transient exception handling is in place
         retryOnConnectionFailure(true)
 
         connectTimeout(config.connectTimeout.toJavaDuration())
         readTimeout(config.socketReadTimeout.toJavaDuration())
         writeTimeout(config.socketWriteTimeout.toJavaDuration())
 
-        // use our own pool configured with the settings taken from config
+        // use our own pool configured with the timeout settings taken from config
         val pool = ConnectionPool(
-            maxIdleConnections = config.maxConnections.toInt(),
+            maxIdleConnections = 5, // The default from the no-arg ConnectionPool() constructor
             keepAliveDuration = config.connectionIdleTimeout.inWholeMilliseconds,
             TimeUnit.MILLISECONDS,
         )
         connectionPool(pool)
 
+        // Configure a dispatcher that uses maxConnections as a proxy for maxRequests. Note that this isn't precisely
```

Comment:
Comment: I don't think we have any better option but do we have any concerns with this imprecise mapping? H2 isn't heavily used yet but this mapping could certainly limit throughput under certain setups.

I did find one of the maintainers [take](https://github.com/square/okhttp/issues/7084#issuecomment-1042872209) on not exposing a proper max connections knob.

Reply:
Ya IDK